src/import/sst_importer.rs

```diff
@@ -0,0 +1,386 @@
+        Ok(SSTImporter {
+            dir: ImportDir::new(root)?,
+            token: AtomicUsize::new(1),
+            files: Mutex::new(HashMap::new()),
+        })
+    }
+
+    pub fn token(&self) -> Token {
+        self.token.fetch_add(1, Ordering::SeqCst)
+    }
+
+    fn insert(&self, token: Token, file: ImportFile) {
+        let mut files = self.files.lock().unwrap();
+        assert!(files.insert(token, file).is_none());
+    }
+
+    pub fn remove(&self, token: Token) -> Option<ImportFile> {
+        let mut files = self.files.lock().unwrap();
+        files.remove(&token)
+    }
+
+    pub fn create(&self, token: Token, meta: &SSTMeta) -> Result<()> {
+        match self.dir.create(meta) {
+            Ok(f) => {
+                info!("create {:?}", f);
+                self.insert(token, f);
+                Ok(())
+            }
+            Err(e) => {
+                error!("create {:?}: {:?}", meta, e);
+                Err(e)
+            }
+        }
+    }
+
+    pub fn append(&self, token: Token, data: &[u8]) -> Result<()> {
+        match self.remove(token) {
+            Some(mut f) => match f.append(data) {
+                Ok(_) => {
+                    self.insert(token, f);
+                    Ok(())
+                }
+                Err(e) => {
+                    error!("append {:?}: {:?}", f, e);
+                    Err(e)
+                }
+            },
+            None => Err(Error::TokenNotFound(token)),
+        }
+    }
+
+    pub fn finish(&self, token: Token) -> Result<()> {
+        match self.remove(token) {
+            Some(mut f) => match f.finish() {
+                Ok(_) => {
+                    info!("finish {:?}", f);
+                    Ok(())
+                }
+                Err(e) => {
+                    error!("finish {:?}: {:?}", f, e);
+                    Err(e)
+                }
+            },
+            None => Err(Error::TokenNotFound(token)),
+        }
+    }
+
+    pub fn ingest(&self, meta: &SSTMeta, db: &DB) -> Result<()> {
+        match self.dir.ingest(meta, db) {
+            Ok(_) => {
+                info!("ingest {:?}", meta);
+                Ok(())
+            }
+            Err(e) => {
+                error!("ingest {:?}: {:?}", meta, e);
+                Err(e)
+            }
+        }
+    }
+}
+
+// TODO: Add size and rate limit.
+pub struct ImportDir {
+    root_dir: PathBuf,
+    temp_dir: PathBuf,
+    clone_dir: PathBuf,
+}
+
+impl ImportDir {
+    const TEMP_DIR: &'static str = ".temp";
+    const CLONE_DIR: &'static str = ".clone";
+
+    fn new<P: AsRef<Path>>(root: P) -> Result<ImportDir> {
+        let root_dir = root.as_ref().to_owned();
+        let temp_dir = root_dir.join(Self::TEMP_DIR);
+        let clone_dir = root_dir.join(Self::CLONE_DIR);
+        if temp_dir.exists() {
+            fs::remove_dir_all(&temp_dir)?;
+        }
+        if clone_dir.exists() {
+            fs::remove_dir_all(&clone_dir)?;
+        }
```

Comment:
Should we clean the root dir first to prevent there are some existing files?

Reply:
`temp_dir` and `clone_dir` can be cleaned up on a restart because they are temporary and their files cannot be used anymore. But `root_dir` contains persisted files which can be ingested later, so we can not remove a file in `root_dir` unless we can make sure that the file will not be ingested anymore. Another PR will do this cleanup job.